simpleclient_eclipselink/src/main/java/io/prometheus/client/eclipselink/EclipseLinkStatisticsCollector.java

```diff
@@ -0,0 +1,744 @@
+    if (sessions.isEmpty()) {
+      throw new IllegalStateException("You must register at least one Session.");
+    }
+    return super.register(registry);
+  }
+
+  private List<MetricFamilySamples> getCounters() {
+    return Arrays.<MetricFamilySamples>asList(
+        createCounter(
+            "eclipselink_uow_commits_total",
+            "Total number of unit of work commits",
+            new ValueProvider() {
+              @Override
+              public double getValue(Map<String, Object> metrics) {
+                Object value = metrics.get(SessionProfiler.UowCommits);
+                if (value != null) {
+                  return ((Long) value).doubleValue();
+                }
+                return 0L;
+              }
+            }
+        ),
+        createCounter(
+            "eclipselink_uow_created_total",
+            "Total number of created units of work",
+            new ValueProvider() {
+              @Override
+              public double getValue(Map<String, Object> metrics) {
+                Object value = metrics.get(SessionProfiler.UowCreated);
+                if (value != null) {
+                  return ((Long) value).doubleValue();
+                }
+                return 0L;
+              }
+            }
+        ),
+        createCounter(
+            "eclipselink_uow_released_total",
+            "Total number of released units of work",
+            new ValueProvider() {
+              @Override
+              public double getValue(Map<String, Object> metrics) {
+                Object value = metrics.get(SessionProfiler.UowReleased);
+                if (value != null) {
+                  return ((Long) value).doubleValue();
+                }
+                return 0L;
+              }
+            }
+        ),
+        createCounter(
+            "eclipselink_uow_rollbacks_total",
+            "Total number of unit of work rollbacks",
+            new ValueProvider() {
+              @Override
+              public double getValue(Map<String, Object> metrics) {
+                Object value = metrics.get(SessionProfiler.UowRollbacks);
+                if (value != null) {
+                  return ((Integer) value).doubleValue();
+                }
+                return 0;
+              }
+            }
+        ),
+        createCounter(
+            "eclipselink_cache_hits_total",
+            "Total number of cache hits",
+            new ValueProvider() {
+              @Override
+              public double getValue(Map<String, Object> metrics) {
+                Object value = metrics.get(SessionProfiler.CacheHits);
+                if (value != null) {
+                  return ((Integer) value).doubleValue();
+                }
+                return 0;
+              }
+            }
+        ),
+        createCounter(
+            "eclipselink_cache_misses_total",
+            "Total number of cache misses",
```

Comment:
It's best to have cache requests as a metric too, it makes it easier to calculate the hit ratio.

Reply:
Agree, but the metric as such was not present in EclipseLink statistics. I can add cache requests as the sum of cache hits and cache misses.